src/main/java/com/dbschema/mongo/MongoDatabaseMetaData.java

```diff
@@ -61,18 +77,18 @@ public ResultSet getTables(String catalogName, String schemaPattern, String tabl
     resultSet.setColumnNames("TABLE_CAT", "TABLE_SCHEM", "TABLE_NAME",
         "TABLE_TYPE", "REMARKS", "TYPE_CAT", "TYPE_SCHEM", "TYPE_NAME", "SELF_REFERENCING_COL_NAME",
         "REF_GENERATION");
-    if (schemaPattern == null) {
-      for (String cat : con.getService().getDatabaseNames()) {
-        for (String tableName : con.getService().getCollectionNames(cat)) {
-          resultSet.addRow(createTableRow(null, tableName));
+    Pattern pSchema = getPattern(schemaPattern);
+    Pattern pTable = getPattern(tableNamePattern);
+
+    for (String schema : con.getService().getDatabaseNames()) {
+      if (schemaPattern == null || schemaPattern.equals(schema) || (pSchema != null && pSchema.matcher(schema).matches())) {
+        for (String tableName : con.getService().getCollectionNames(schema)) {
+          if (tableNamePattern == null || tableNamePattern.equals(tableName) || (pTable != null && pTable.matcher(tableName).matches()))
```

Comment:
I guess if we compile pattern then we should not check `tableNamePattern.equals(tableName)`

Reply:
So as suggested added support for getSearchStringEscape.  I also pulled the getPattern() method into a separate class as it should really be used in a few more spots - for example getPrimaryKeys (with tableNamePattern).  I also added a separate class for the associated junit tests.